overlapfinder_melvin.py

```diff
@@ -0,0 +1,130 @@
+from datetime import datetime as dt
+from intervaltree import Interval, IntervalTree
+#import test_inputs
+
+# one interval
+tc_0 = [Interval(1, 3, "aaron")]
+
+# no overlap
+tc_1 = [Interval(1, 2, "aaron"), Interval(3, 4, "ben"), Interval(5, 6, "charlotte")]
+
+# separate overlaps
+tc_2 = [Interval(1, 3, "aaron"), Interval(2, 4, "ben"),  Interval(3, 5, "charlotte")]
+
+# shared overlap
+tc_3 = [Interval(1, 3, "aaron"), Interval(2, 4, "ben"), Interval(2, 5, "charlotte")]
+
+# 4 intervals
+tc_4 = [Interval(1, 3, "aaron"), Interval(2, 4, "ben"),
+        Interval(3, 5, "charlotte"), Interval(1, 6, "dan")]
+
+
+
+# Input: An list of Intervals
+# Output: A dictionary (key: overlap, value: set of user_ids which share the overlap)
+def find_all_common_intervals(interval_list):
+    overlap_dict = dict()
+    interval_tree = IntervalTree(interval_list)
+    
+    for interval in interval_tree.items(): # create a copy of the tree to iterate over
+        # check interval against overlap_dict
+        for overlap in overlap_dict.copy():
```

Comment:
Can use `overlap_dict.keys()` instead of `overlap_dict.copy()`?

Not logically consequential but makes code clearer imho.